go/worker/common/p2p/p2p.go

```diff
@@ -212,9 +222,15 @@ func New(ctx context.Context, identity *identity.Identity, consensus consensus.B
 		return nil, fmt.Errorf("worker/common/p2p: failed to initialize libp2p gossipsub: %w", err)
 	}
 
+	doc, err := consensus.GetGenesisDocument(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("worker/common/p2p: failed to get consensus genesis document: %w", err)
+	}
+
 	p := &P2P{
 		PeerManager:       newPeerManager(ctx, host, consensus),
 		ctx:               ctx,
+		chainID:           doc.ChainID,
```

Comment:
This should use `doc.ChainContext()` instead (and maybe call the field `chainContext`).